scripts/build_shapes.py

```diff
@@ -80,7 +81,64 @@ def download_GADM(country_code, update=False, out_logging=False):
     return GADM_inputfile_gpkg, GADM_filename
 
 
-def get_GADM_layer(country_list, layer_id, geo_crs, update=False, outlogging=False):
+def replace_nonstd_codes(row, col, country_code, keep_cond=True):
+    if row["GID_0"] != country_code:
+        return country_name_2_two_digits(row["COUNTRY"])
+    else:
+        return row["GID_0"]
+
+
+def filter_gadm(geodf, layer, cc, output_nonstd_to_csv=True, keep_all_areas=True):
+
+    # convert country name representation of the main country (GID_0 column)
+    geodf["GID_0"] = geodf["GID_0"].map(three_2_two_digits_country)
+
+    if keep_all_areas:
+        # in case GID_0 have any exotic values, "COUNTRY" column may be used instead
+        geodf["GID_0"] = geodf.apply(
+            lambda x: replace_nonstd_codes(x, col="GID_0", country_code=cc), axis=1
+        )
+        # "not found" hardcoded according to country_converter conventions
+        geodf.drop(geodf[geodf["GID_0"] == "not found"].index, inplace=True)
```

Comment:
I think lines from 98 to 108 may be replaced with the following if GID_0 supposed to be the 3 country letter code?
``` python

geodf.drop(geodf[geodf["GID_0"] != two_2_three_digits_country(cc)].index, inplace=True)
```
Calling the conversion function may take some time that may be avoided, unless necessary.

What do you think?

Reply:
But it would drop all the non-standard codes from the very beginning instead trying to restore them. What do you mean as a conversion function?

Reply:
The comment was linked to the revision described below in which the first if would take care of the option drop.

In programming a typical approach is to manage the cases from the most particular (in our case drop) with ifs-elifs to the default value (in our case set_by_country). That is a clean way to do that.

Think about the first case: what if a user adds an additional case to the if list?
We should revise the first if into something like option not in list, where list contains the options that follow. That is not so clean.
Often these conditions are infact managed like:
if a [first particular]
elif b [second particular]
else c [the default case]

If a user is willing to add an additional option, simply an additional elif is added and anything else

Reply:
That absolutely makes sense. Thanks a lot for the explanation!